public/app/plugins/datasource/cloudwatch/config_ctrl.ts

```diff
@@ -36,4 +39,23 @@ export class CloudWatchConfigCtrl {
     { name: 'Monthly', value: 'Monthly', example: '[logstash-]YYYY.MM' },
     { name: 'Yearly', value: 'Yearly', example: '[logstash-]YYYY' },
   ];
+
+  regions = [];
+
+  getRegions() {
+    this.$http.get('https://ip-ranges.amazonaws.com/ip-ranges.json').then(ip_ranges => {
```

Comment:
can't we use the same backend implementation here?

Reply:
sorry, see that this will be tricky as the datasource has yet to be added

Reply:
Using `ec2:DescribeRegions` is most appropriate way to get regions.
https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRegions.html
But it is impossible to call it before creating datasource. (Sorry I should explain it)

The region list can get from `ip-ranges.json` too.
https://docs.aws.amazon.com/general/latest/gr/aws-ip-ranges.html
It is 167KB file, I think we shouldn't access the `ip-ranges.json` so often.
We can't use it for templating.

Alternative way is getting `ip-ranges.json` in backend code periodically, and cache region list.
But, I think adding timer for this simple thing is not good, so current implementation is most simple and best.

Reply:
Would it be possible to use similar approach as in graphite datasource which auto detects the version after the datasource have been saved? See [code](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/graphite/config_ctrl.ts#L11).

Reply:
Looks like a good approach! I change the code.

Reply:
Or maybe not, having another [discussion](https://github.com/grafana/grafana/pull/12680#discussion_r204424343) regarding this right now. You'll still end up with the problem where you want the user to select a region after datasource have been saved first time and then the user needs to save the datasource again.